SwiftGit2/Errors.swift

```diff
@@ -3,27 +3,29 @@ import libgit2
 
 public let libGit2ErrorDomain = "org.libgit2.libgit2"
 
-/// Returns an NSError with an error domain and message for libgit2 errors.
-///
-/// :param: errorCode An error code returned by a libgit2 function.
-/// :param: libGit2PointOfFailure The name of the libgit2 function that produced the
-///         error code.
-/// :returns: An NSError with a libgit2 error domain, code, and message.
-internal func libGit2Error(_ errorCode: Int32, libGit2PointOfFailure: String? = nil) -> NSError {
-	let code = Int(errorCode)
-	var userInfo: [String: String] = [:]
+internal extension NSError {
```

Comment:
This could probably be a custom error type now. I don't think the functionality was implemented when I started SwiftGit2.

(But I also don't think that's required for this PR.)